diff --git a/src/Security/OpenIdConfigurationProvider.php b/src/Security/OpenIdConfigurationProvider.php
index bf96661..d195442 100644
--- a/src/Security/OpenIdConfigurationProvider.php
+++ b/src/Security/OpenIdConfigurationProvider.php
@@ -541,10 +541,6 @@ class OpenIdConfigurationProvider extends AbstractProvider
             throw new BadUrlException('OpenIDConnectMetadataUrl is invalid: ' . $url);
         }
 
-        if (parse_url($url, PHP_URL_SCHEME) !== 'https') {
-            throw new IllegalSchemeException('OpenIDConnectMetadataUrl must use https: ' . $url);
-        }
-
         $this->openIDConnectMetadataUrl = $url;
     }
 }
