diff --git a/maestro.module b/maestro.module
index c212dc0..e4e0a52 100644
--- a/maestro.module
+++ b/maestro.module
@@ -9,6 +9,7 @@ use Drupal\webform\Entity\WebformSubmission;
 use Drupal\maestro\Utility\TaskHandler;
 use Drupal\Core\Render\BubbleableMetadata;
 use Drupal\Component\Utility\Html;
+use Drupal\Component\Utility\NestedArray;
 use Drupal\Core\Url;
 use Drupal\Core\Entity\Display\EntityViewDisplayInterface;
 use Drupal\Core\Entity\EntityInterface;
@@ -597,6 +598,11 @@ function maestro_token_info() {
     'description' => t('Use the Entity Identifier to pull a value out of a field.  example: maestro:entity-identifier:uniqueid:field_machine_name.'),
   ];
 
+  $maestro['task'] = [
+    'name' => t('Task'),
+    'description' => t('Use the Task to pull a value out of the template task definition.  example: maestro:task:data:*.'),
+  ];
+
   return [
     'types' => ['maestro' => $type],
     'tokens' => [
@@ -749,6 +755,11 @@ function maestro_tokens($type, $tokens, array $data, array $options, BubbleableM
                 }
               }
             }
+            elseif ($token_parts[0] == 'task') {
+              array_shift($token_parts);
+              // Get the task value and fall back to the empty string.
+              $replace = NestedArray::getValue($task, $token_parts) ?: '';
+            }
           }
           $replacements[$original] = $replace;
           break;
