diff --git a/modules/webform_revisions/webform_revisions.install b/modules/webform_revisions/webform_revisions.install
index 8177c70..801e68f 100644
--- a/modules/webform_revisions/webform_revisions.install
+++ b/modules/webform_revisions/webform_revisions.install
@@ -9,12 +9,16 @@ function webform_revisions_install() {
   $entityTypeManager = \Drupal::entityTypeManager();
   $cert_storage = $entityTypeManager->getStorage('config_entity_revisions_type');
 
-  $entity = $cert_storage->create([
-    'id' => 'webform_revisions',
-    'label' => 'Webform Revisions',
-  ]);
-
-  $entity->save();
+  // This hook_install() implementation is potentially called twice when
+  // importing config during `drush site:install --with-existing-config`.
+  if (null === $cert_storage->load('webform_revisions')) {
+    $entity = $cert_storage->create([
+      'id' => 'webform_revisions',
+      'label' => 'Webform Revisions',
+    ]);
+
+    $entity->save();
+  }
 }
 
 /**
